site/examples/code-highlighting.js

```diff
@@ -0,0 +1,164 @@
+          return token.content.length
+        } else {
+          return token.content.reduce((l, t) => l + getLength(t), 0)
+        }
+      }
+      const tokens = Prism.tokenize(node.text, Prism.languages[language])
+      let start = 0
+
+      for (const token of tokens) {
+        const length = getLength(token)
+        const end = start + length
+
+        if (typeof token !== 'string') {
+          ranges.push({
+            [token.type]: true,
+            anchor: { path, offset: start },
+            focus: { path, offset: end },
+          })
+        }
+
+        start = end
+      }
+
+      return ranges
+    },
+    [language]
+  )
+
+  return (
+    <Slate editor={editor} value={value} onChange={value => setValue(value)}>
+      <div
+        contentEditable={false}
+        style={{ position: 'relative', top: '5px', right: '5px' }}
+      >
+        <h3>
+          Select a language
+          <select
+            value={language}
+            style={{ float: 'right' }}
+            onChange={e => setLanguage(e.target.value)}
+          >
+            <option value="js">JavaScript</option>
+            <option value="css">CSS</option>
+            <option value="html">HTML</option>
+            <option value="python">Python</option>
+            <option value="sql">SQL</option>
+            <option value="java">Java</option>
+            <option value="php">PHP</option>
+          </select>
+        </h3>
+      </div>
+      <Editable
+        decorate={decorate}
+        renderLeaf={renderLeaf}
+        placeholder="Write some code..."
+      />
+    </Slate>
+  )
+}
+
+// different token types, styles found on Prismjs website
+const Leaf = ({ attributes, children, leaf }) => {
+  return (
+    <span
+      {...attributes}
+      className={css`
+            font-family: monospace;
+            background: hsla(0, 0%, 100%, .5);
+
+        ${leaf.comment &&
+          css`
+            color: slategray;
+          `} 
+
+        ${(leaf.operator || leaf.url) &&
+          css`
+            color: #9a6e3a;
+          `}
+        ${leaf.keyword &&
+          css`
+            color: #07a;
+          `}
+        ${(leaf.variable || leaf.regex) &&
+          css`
+            color: #e90;
+          `}
+        ${(leaf.number ||
+          leaf.boolean ||
+          leaf.tag ||
+          leaf.constant ||
+          leaf.symbol ||
+          leaf.attr - name ||
+          leaf.selector) &&
+          css`
+            color: #905;
+          `}
+        ${leaf.punctuation &&
+          css`
+            color: #999;
+          `}
+        ${(leaf.string || leaf.char) &&
+          css`
+            color: #690;
+          `}
+        ${(leaf.function || leaf.class - name) &&
+          css`
+            color: #dd4a68;
+          `}
+        `}
+    >
+      {children}
+    </span>
+  )
+}
+
+const initialValue = [
+  {
+    children: [
+      {
+        text: '<h1>Hi!</h1>',
+      },
+    ],
+  },
+]
+
```

Comment:
As mentioned above, the `getLength` function can be extracted outside of the component and defined down here.

```suggestion

function getLength(token) {
  if (typeof token === 'string') return token.length
  if (typeof token.content === 'string') return token.content.length

  return token.content.reduce((l, t) => l + getLength(t), 0)
}

```

Also note that we don't need `else` if we are `return`-ing on the previous line.